tests/test_project.py

```diff
@@ -562,6 +563,70 @@ def test_init_local_missing_west_yml_failure(repos_tmpdir):
         cmd(f'init -l "{zephyr_install_dir}"')
 
 
+def test_init_local_with_manifest_filename(repos_tmpdir):
+    # Test 'west init --mf -l' on a local repo
+
+    manifest = repos_tmpdir / 'repos' / 'zephyr'
+    workspace = repos_tmpdir / 'workspace'
+    zephyr_install_dir = workspace / 'zephyr'
+
+    # Do a local clone of manifest repo
+    clone(str(manifest), str(zephyr_install_dir))
+    os.rename(str(zephyr_install_dir / 'west.yml'),
+              str(zephyr_install_dir / 'project.yml'))
+
+    # fails because west.yml is missing
+    with pytest.raises(subprocess.CalledProcessError):
+        cmd(f'init -l "{zephyr_install_dir}"')
+
+    with open(zephyr_install_dir / 'west.yml', 'w') as f:
+        f.write('[')
+
+    # fails because of the syntax error
+    cwd = os.getcwd()
+
+    with pytest.raises(subprocess.CalledProcessError):
+        cmd(f'init -l "{zephyr_install_dir}"')
+        # init with a local manifest doesn't parse the file, so let's access it
+        workspace.chdir()
+        cmd('list')
+
+    os.chdir(cwd)
+    shutil.rmtree(workspace / '.west')
```

Comment:
I would just hack around the windows issue by using `shutil.move()` to put them somewhere harmless inside the tmpdir, instead of `shutil.rmtree()`.